tutorials/preprocessing/plot_40_artifact_correction_ica.py

```diff
@@ -1,307 +1,513 @@
 #
-# .. note:: The default method in MNE is FastICA, which along with Infomax is
-#           one of the most widely used ICA algorithms. Picard is a
-#           new algorithm that is expected to converge faster than FastICA and
-#           Infomax, especially when the aim is to recover accurate maps with
-#           a low tolerance parameter, see [1]_ for more information.
+# See the Notes section of the :class:`~mne.preprocessing.ICA` documentation
+# for further details. Next we'll walk through an extended example that
+# illustrates each of these steps in greater detail.
+#
+#
+# Example: EOG and ECG artifact repair
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# Let's begin by visualizing the artifacts that we want to repair. In this
+# dataset they are big enough to see easily in the raw data:
 
-method = 'fastica'
+# plot some channels that clearly show heartbeats and blinks
+regexp = r'(MEG [12][45][123]1|EEG 00.)'
+artifact_picks = mne.pick_channels_regexp(raw.ch_names, regexp=regexp)
+raw.plot(order=artifact_picks, n_channels=len(artifact_picks))
 
-# Choose other parameters
-n_components = 25  # if float, select n_components by explained variance of PCA
-decim = 3  # we need sufficient statistics, not all time points -> saves time
+###############################################################################
+# We can get a better view of the ocular artifacts using
+# :func:`~mne.preprocessing.create_eog_epochs` like we did in the
+# :ref:`tut-artifact-overview` tutorial:
 
-# we will also set state of the random number generator - ICA is a
-# non-deterministic algorithm, but we want to have the same decomposition
-# and the same order of components each time this tutorial is run
-random_state = 23
+eog_evoked = create_eog_epochs(raw).average()
+eog_evoked.plot_joint()
 
 ###############################################################################
-# Define the ICA object instance
-ica = ICA(n_components=n_components, method=method, random_state=random_state)
-print(ica)
+# Now we'll do the same for the heartbeat artifacts, using
+# :func:`~mne.preprocessing.create_ecg_epochs`:
 
-###############################################################################
-# we avoid fitting ICA on crazy environmental artifacts that would
-# dominate the variance and decomposition
-reject = dict(mag=5e-12, grad=4000e-13)
-ica.fit(raw, picks='meg', decim=decim, reject=reject)
-print(ica)
+ecg_evoked = create_ecg_epochs(raw).average()
+ecg_evoked.plot_joint()
 
 ###############################################################################
-# Plot ICA components
-ica.plot_components()  # can you spot some potential bad guys?
+# Before we run the ICA, an important step is filtering the data to remove
+# low-frequency drifts, which can negatively affect the quality of the ICA fit.
+# A 1 Hz cutoff frequency is recommended; we'll keep a copy of the unfiltered
+# :class:`~mne.io.Raw` object around so we can apply the ICA solution to it
+# later.
```

Comment:
here it would be good to have a brief explanation for **why** this is possible. It's counter-intuitive to many people how you can estimate an ICA on processed data and then validly apply to to data that has been processed differently.

See this thread from our mailing list: https://mail.nmr.mgh.harvard.edu/pipermail/mne_analysis/2018-March/004785.html